client/task_runner.go

```diff
@@ -446,8 +447,8 @@ func (r *TaskRunner) run() {
 		destroyed := r.destroy
 		r.destroyLock.Unlock()
 		if destroyed {
-			r.logger.Printf("[DEBUG] client: Not restarting task: %v because it's destroyed by user", r.task.Name)
-			r.setState(structs.TaskStateDead, structs.NewTaskEvent(structs.TaskKilled))
+			r.logger.Printf("[DEBUG] client: Not restarting task: %v because it has been destroyed due to: %s", r.task.Name, r.destroyEvent.Message)
+			r.setState(structs.TaskStateDead, r.destroyEvent)
```

Comment:
Sorry you can actually revert this back to what it was originally. You can only get here from a user kill